WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -576,33 +753,14 @@ extension MediaLibraryViewController: WPMediaPickerViewControllerDelegate {
     }
 
     func makeAndUploadMediaWith(_ asset: PHAsset) {
-        let assetID = asset.identifier()
-
+        createMediaCompletionHander = uploadMedia
         let service = MediaService(managedObjectContext: ContextManager.sharedInstance().mainContext)
         service.createMedia(with: asset,
                             forBlogObjectID: blog.objectID,
                             thumbnailCallback: nil,
                             completion: { [weak self] media, error in
-                                guard let media = media else {
-                                    if let error = error as NSError? {
-                                        self?.mediaProgressCoordinator.attach(error: error, toMediaID: assetID)
-                                    }
-                                    return
-                                }
-
-                                var uploadProgress: Progress? = nil
-                                service.uploadMedia(media, progress: &uploadProgress, success: { [weak self] in
-                                    self?.unpauseDataSource()
-                                    self?.trackUploadFor(media)
-                                }, failure: { error in
-                                    self?.mediaProgressCoordinator.attach(error: error as NSError, toMediaID: assetID)
-
-                                    self?.unpauseDataSource()
-                                })
-
-                                if let progress = uploadProgress {
-                                    self?.mediaProgressCoordinator.track(progress: progress, ofObject: media, withMediaID: assetID)
-                                }
+                                self?.createMediaCompletionHander!(media, error, asset.identifier())
```

Comment:
I'd probably use a `?` instead of `!` here.

I'm not sure I'm following, but how come you're using a property for the handler here instead of just calling the uploadMedia method directly? I don't see it ever being set to anything else?

Reply:
Because apparently I had some weird desire to make it complicated. It shall be changed!